### PR TITLE
feat(): add default listener functionality

### PR DESCRIPTION
**Pull request by jmaher409**
Added new static and instance setDefaultListener methods
to allow signal creators to set default listeners to be used
when no other listener is added. This should typically be used
to report errors or log when no listener is added as it can
be considered a programmer error in some use cases.

### CODE REVIEW

File: src/signal.ts [hunk 1]
```diff
@@ -1,10 +1,20 @@
...
-import { ReadableSignal } from './index';
+import {ReadableSignal} from './index';
+import {Listener, WritableSignal} from './interfaces';
 
 export class Signal<T> extends ExtendedSignal<T> implements WritableSignal<T>, ReadableSignal<T> {
+    public static setDefaultListener(listener: Listener<any>) {
+        this._staticDefaultListener = listener;
+    }
+
+    protected static _staticDefaultListener: Listener<any> = () => {
```
Comment (lelandmiller) [inline src/signal.ts]: Could this be private instead of protected?

File: src/signal.ts [hunk 2]
```diff
@@ -1,10 +1,20 @@
-import {Listener, WritableSignal} from './interfaces';
-
 import {ExtendedSignal} from './extended-signal';
-import { ReadableSignal } from './index';
+import {ReadableSignal} from './index';
+import {Listener, WritableSignal} from './interfaces';
 
 export class Signal<T> extends ExtendedSignal<T> implements WritableSignal<T>, ReadableSignal<T> {
+    public static setDefaultListener(listener: Listener<any>) {
+        this._staticDefaultListener = listener;
```
Comment (lelandmiller) [inline src/signal.ts]: I don't know why, but in cases like this I usually do Signal._staticDefaultListener. Either works, the difference would be if the context were not set. I guess it doesn't really matter, just never thought to do it this way for some reason :-)

**Comment by lelandmiller** [comment]
Would it also be possible to add an entry to the README?

**Comment by jmaher409** [comment]
Done.

**Pull request merged**